scipy/stats/_fit.py

```diff
@@ -28,13 +28,14 @@ def _combine_bounds(name, user_bounds, shape_domain, integral):
                    "of the distribution.")
         raise ValueError(message)
 
+    # Only if there is no analytical fit are we concerned about infinite bounds
+    message = None
     if not np.all(np.isfinite(bounds)):
         message = (f"The intersection of user-provided bounds for `{name}` "
-                   f"and the domain of the distribution is not finite. Please "
-                   f"provide finite bounds for shape `{name}` in `bounds`.")
-        raise ValueError(message)
+                    f"and the domain of the distribution is not finite. Please "
+                    f"provide finite bounds for shape `{name}` in `bounds`.")
 
-    return bounds
+    return bounds, message
```

Comment:
Check whether there is a distribution-specific `fit` method before complaining that the user hasn't provided bounds. If there is a distribution-specific `fit` method, we'll try it. If it works, we'll return the result. If not, we'll raise the error message, because we'd need bounds to do the general fit procedure.